antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -1050,3 +1051,35 @@ def initialize_additional_data(self, variant_study: VariantStudy) -> bool:
                 exc_info=e,
             )
             return False
+
+    def clear_all_snapshots(self, limit: int, params: t.Optional[RequestParameters] = None) -> None:
+        """
+        Clear all variant snapshots older than `limit` (in hours).
+        Only available for admin users.
+        `limit` must be a positive integer.
+        Raises a UserHasNotPermissionError if the user is not administrator
+        Args:
+            limit (integer): number of hours
+            params: request parameters used to identify the user status
+        Returns: None
+
+        Raises:
+            UserHasNotPermissionError
+            HTTPException
+        """
+        if limit < 0:
+            raise HTTPException(status_code=400, detail=f"Limit cannot be negative (limit={limit})")
+        if params.user.is_site_admin():
+            result = self.repository.get_all(
+                study_filter=StudyFilter(
+                    variant=True,
+                    access_permissions=AccessPermissions(is_admin=True),
+                )
+            )
+            for variant in result:
+                if variant.updated_at < datetime.now() - timedelta(hours=limit) or (
```

Comment:
Maybe it's better to have an "and" condition: if it has been modified or read recently, we don't delete it.
For example, if it has been read recently, although it has not been modified, it makes sense to keep it.